components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitMQComponent.java

```diff
@@ -202,6 +204,7 @@ protected RabbitMQEndpoint createEndpoint(String uri,
         endpoint.setAddresses(getAddresses());
         endpoint.setThreadPoolSize(getThreadPoolSize());
         endpoint.setExchangeName(exchangeName);
+        endpoint.setArgs(args);
```

Comment:
Same here, I think this is not needed, as this being handled by `setProperties(endpoint, params);`

Reply:
This is all the point params parameter is always an empty map where args is setting using query parameters is my PR there is something wrong i don't know where i think we have to cleanup this component

Reply:
What do you mean? What are you trying to fix exactly here? Because I feel we are mixing things up here.

Reply:
I trying to first getting args argument when it passed as query parameters lets just fix this for beginning

Reply:
But I thought if you pass it like this `rabbitmq:foo?queue=foo&arg.queue.x-max-priority=10` it will work?

Reply:
Yes is work but as i explain in my ticket how i suppose to do with the camelenpointdsl to achieve the same this is not a regular way to pass query parameter and now we see the limitation

Reply:
So maybe is better to rollback all this stuff and restore the oldest and good way using some param that take a map and then use this eg:params=#paramsMap

Reply:
Okay, how do you call your rabbitmq then via endpoint dsl?

Reply:
Like your example could be a good way i take your example from the ticket:

`rabbitmq("foo")
        .queue("foo")
        .advanced()
        .args(Collections.singletonMap("queue.x-max-priority", "10"));`

Reply:
Okay, based on this one, how about if you just remove the lines that I suggested and try it out.

Reply:
What do you mean ? i really didn't understand

Reply:
The case is simple how i suppose to translate this weird way `&arg.queue.x-max-priority=10` of passing parameter with the new DSL

Reply:
In this PR, I suggested to remove some lines from the code. This line where we are commenting now and [here](https://github.com/apache/camel/pull/3385#discussion_r354166344)

Reply:
But this is the main point if your remove this line i'll get an empty map

Reply:
Maybe if you can try this and i think you'll see what i'm trying to solve